packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visit-summary.component.tsx

```diff
@@ -0,0 +1,135 @@
+        medications = medications.concat(
+          enc.orders.map((order: Order) => ({
+            order,
+            provider: {
+              name: enc.encounterProviders.length ? enc.encounterProviders[0].provider.person.display : '',
+              role: enc.encounterProviders.length ? enc.encounterProviders[0].encounterRole.display : '',
+            },
+          })),
+        );
+
+        // Check for Visit Diagnoses and Notes
+        if (enc.encounterType.display === 'Visit Note') {
+          enc.obs.forEach((obs: Observation) => {
+            if (obs.concept.display === 'Visit Diagnoses') {
+              // Putting all the diagnoses in a single array.
+              diagnoses.push({
+                diagnosis: obs.groupMembers.find((mem) => mem.concept.display === 'PROBLEM LIST').value.display,
+                order: obs.groupMembers.find((mem) => mem.concept.display === 'Diagnosis order').value.display,
+              });
+            } else if (obs.concept.display === 'Text of encounter note') {
+              // Putting all notes in a single array.
+              notes.push({
+                note: obs.value,
+                provider: {
+                  name: enc.encounterProviders.length ? enc.encounterProviders[0].provider.person.display : '',
+                  role: enc.encounterProviders.length ? enc.encounterProviders[0].encounterRole.display : '',
+                },
+                time: formatTime(obs.obsDatetime),
+              });
+            }
+          });
+        } else {
+          enc.obs.forEach((obs: Observation) => {
+            if (obs.concept.conceptClass.display === 'Test' || obs.concept.conceptClass.display === 'LabSet')
+              tests.push({
+                testName: obs.concept.display,
+                value: obs.value,
+              });
+          });
+        }
+      });
+      return [diagnoses, notes, tests, medications];
+    }, [encounters]);
+
+  return (
+    <div className={styles.summaryContainer}>
+      <p className={styles.productiveHeading01}>{t('diagnoses', 'Diagnoses')}</p>
+      <div className={`${styles.caption01} ${styles.diagnosesList}`}>
+        {diagnoses.length > 0 ? (
+          diagnoses.map((d: DiagnosisItem, ind) => (
+            <span
+              key={ind}
+              className={`${styles.diagnosis} ${
+                d.order === 'Primary' ? styles.primaryDiagnose : styles.secondaryDiagnose
+              }`}>
+              {d.diagnosis}
+            </span>
+          ))
+        ) : (
+          <span className={`${styles.bodyLong01} ${styles.text02}`} style={{ marginBottom: '0.5rem' }}>
+            {t('noDiagnosesFound', 'No diagnoses found')}
+          </span>
+        )}
+      </div>
+      <Tabs className={styles.verticalTabs}>
+        <Tab
+          className={`${styles.tab} ${styles.bodyLong01} ${tabSelected === 0 && styles.selectedTab}`}
+          onClick={() => setSelectedTab(0)}
+          id="tab-1"
```

Comment:
The tab ids a quite generic here. Could we make them a bit more specific? Not sure if this has any impact anyway...

Reply:
Done